code/modules/projectiles/guns/projectile/constructable/blastcannon.dm

```diff
@@ -17,10 +18,14 @@
 	conventional_firearm = 0
 	silenced = 1
 	var/obj/item/device/transfer_valve/bomb = null
+	var/obj/item/toy/bomb/toybomb = null
 	var/datum/gas_mixture/bomb_air_contents_1 = null
 	var/datum/gas_mixture/bomb_air_contents_2 = null
 	var/ignorecap = 0
+	var/overcap = 0
 	var/bomb_appearance = null
+	var/widening_rate = 5
```

Comment:
this whole widening cone revamp to the blast cannon is such a significant change that it really should be its own PR for atomicity, but no one seems to be objecting to it so maybe it's fine

Perhaps people just don't see blast cannons enough these days but they were very controversial once upon a time, and I feel like people might have missed this change in the large gamut of other changes.

Reply:
People were very joyous about it when I displayed the blast canon changes on Discord. If it proves to be a bad idea in hindsight, simply editing that var to something like 999 will cause the blast canon to fire in a thin line again.

Reply:
Also worth noting, they might have been controversial due to ignoring the explosion blocking effect of walls and other obstacles, which the new code takes into account.